tests/unittests/tests-unicoap/tests-unicoap-message.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (C) 2024-2025 Carl Seifert
+ * Copyright (C) 2024-2025 TU Dresden
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @file
+ * @ingroup unittests
+ * @brief   Unit tests for contiguous and noncontiguous payload functionality
+ * @author  Carl Seifert <carl.seifert1@mailbox.tu-dresden.de>
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "tests-unicoap.h"
+
+#include "net/unicoap/message.h"
+#include "net/unicoap/options.h"
+
+static void test_contiguous_payload(void)
+{
+    unicoap_message_t message = { .method = UNICOAP_METHOD_GET };
+
+    TEST_ASSERT_EQUAL_INT(unicoap_message_payload_is_empty(&message), true);
```

Comment:
```suggestion
    TEST_ASSERT(unicoap_message_payload_is_empty(&message));
```

and probably in many other places.